tests/functional_tests/fencil_processor_tests/codegen_tests/test_gtfn.py

```diff
@@ -57,15 +60,15 @@ def fencil_example():
         ],
     )
     params = [
-        source_modules.BufferParameter("buf", ["I"], numpy.dtype(numpy.float32)),
-        source_modules.ScalarParameter("sc", numpy.dtype(numpy.float32)),
+        np_as_located_field(Dimension("I"))(numpy.empty((1,), dtype=numpy.float32)),
```

Comment:
I think it would be better to first declare `IDim = Dimension("I")` to stick to the conventions.

I don't know the semantics of `Dimension("D") == Dimension("D")`. Is the identity tied to the object or to the string?

Reply:
`Dimension` is a frozen dataclass with `.value` and a `.kind` attributes. By default, dataclasses generate an `__eq__` method which compares the values of the attributes. In the above example the identity is tied to the string, not the object's address in memory.

From the [dataclass docs](https://docs.python.org/3/library/dataclasses.html?highlight=dataclass#dataclasses.dataclass):
> eq: If true (the default), an __eq__() method will be generated. This method compares the class as if it were a tuple of its fields, in order. Both instances in the comparison must be of the identical type.